buildSrc/src/main/java/quilt/internal/tasks/lint/SpellingChecker.java

```diff
@@ -0,0 +1,109 @@
+package quilt.internal.tasks.lint;
+
+import cuchaz.enigma.translation.mapping.EntryMapping;
+import cuchaz.enigma.translation.representation.AccessFlags;
+import cuchaz.enigma.translation.representation.entry.ClassEntry;
+import cuchaz.enigma.translation.representation.entry.Entry;
+import cuchaz.enigma.translation.representation.entry.MethodEntry;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class SpellingChecker implements Checker<Entry<?>> {
+    private static final String PACKAGE_INFO_CLASS_PACKAGE = "net/minecraft/unused/packageinfo/";
+
+    private static final Set<String> ALLOWED_WORDS = new HashSet<>();
+    static {
+        // collect allowed words
+        URL url;
+        try {
+            url = new URL("https://raw.githubusercontent.com/ix0rai/qm-base-allowed-wordlist/main/allowed_english_words.txt");
+            Set<String> englishWords = new java.util.Scanner(url.openStream()).useDelimiter("\\A").next().lines().collect(Collectors.toSet());
+            ALLOWED_WORDS.addAll(englishWords);
+        } catch (Exception e) {
+            throw new RuntimeException("failed to download and read english word file for spell check in mapping lint task!");
+        }
+
+        // collect minecraft words
+        InputStream parsedFile = SpellingChecker.class.getClassLoader().getResourceAsStream("minecraft_specific_words.txt");
+        if (parsedFile == null) {
+            throw new RuntimeException("could not find minecraft word file for spelling check in mapping lint task!");
+        }
+
+        Set<String> minecraftWords = getLines(parsedFile);
+        // ignore comments and empty lines
+        Set<String> minecraftWordsCopy = Set.copyOf(minecraftWords);
+        for (String word : minecraftWordsCopy) {
+            if (word.isEmpty() || word.startsWith("//")) {
+                minecraftWords.remove(word);
+            }
+        }
+
+        ALLOWED_WORDS.addAll(minecraftWords);
+    }
+
+    private static Set<String> getLines(InputStream stream) {
+        return new java.util.Scanner(stream).useDelimiter("\\A").next().lines().collect(Collectors.toSet());
```

Comment:
You could use `new BufferedReader(new InputStreamReader(stream)).lines().collect(Collectors.toSet())` instead of a Scanner. You can also filter out comment lines with `.filter(s -> !s.startsWith("//"))` before `.collect`